dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +123,209 @@ async def get_cluster_status(self, req):
                 cluster_status=debug_status(formatted_status_string, error),
             )
 
+    """
+    We don't use pid and ip to get task traceback since ip and pid is bounded to a worker
+    and a worker may run different tasks at different time.
+    Therefore, we may provide wrong traceback info since we don't know which task is running on the worker
+
+    Raises:
+        ValueError: HTTPInternalServerError
+    """
+
+    @routes.get("/task/traceback")
```

Comment:
@alanwguo , this is the prototype of the API. I noticed that there is some duplicate code in both @routes.get("/task/traceback") and @routes.get("/task/cpu_profile"). Do you have any suggestions on how to handle this duplication, or should we keep it as it is?

Reply:
we should do a function for `get_worker_details_for_running_task` which returns pid and ip and current state of the task.

Then we can do 
```
task_state, pid, ip = get_worker_details_for_task(task_id, attempt_number)
if task_state is not running:
  raise Error()
result = get_traceback(pid, ip) # or cpu_profiling()
task_state, _ = get_worker_details_for_task(task_id, attempt_number)
if task_state is not running:
  raise Error()
return result
```

Reply:
Sounds good!